From 176f114cff480c9a80d7f11c2f408bdc50be2be6 Mon Sep 17 00:00:00 2001
From: Andi Kleen <ak@linux.intel.com>
Date: Tue, 7 Jul 2020 16:15:18 -0700
Subject: [PATCH 29/39] x86/purgatory/lto: Disable gcc LTO for purgatory

Various issues with gcc LTOing the purgatory code, so disable LTO here.

Signed-off-by: Andi Kleen <ak@linux.intel.com>
---
 arch/x86/purgatory/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/purgatory/Makefile b/arch/x86/purgatory/Makefile
index 95ea17a9d20c..e3b04a62df2d 100644
--- a/arch/x86/purgatory/Makefile
+++ b/arch/x86/purgatory/Makefile
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 OBJECT_FILES_NON_STANDARD := y
+KBUILD_CFLAGS	     += $(DISABLE_LTO_GCC)
 
 purgatory-y := purgatory.o stack.o setup-x86_$(BITS).o sha256.o entry64.o string.o
 
-- 
2.30.2

